### DISCUSSION TITLE
I want to get the coordinate values on the map from the coordinate values of the feature values of each frame.

### DISCUSSION THREAD

**Discussion by yyyutooo** [stella-cv/stella_vslam#154] [Q&A] [answered]
Hello.
I would like to know, from the coordinates of the features of each frame obtained when running OpenVSLAM, which coordinates those features correspond to on the map.I was able to get the coordinates of the features for each frame in the image by asking a question before.However, we do not know which points (coordinates) these features correspond to on the map.
Which code in OpenVSLAM can I refer to to get the coordinate values on the map?
I am sorry for the confusing text, but please help me!
![Untitled 1](https://user-images.githubusercontent.com/70961161/131628480-3b577a4a-9d60-4cd0-afb0-8c3f2fe98391.jpg)

**Comment by ymd-stella** [comment, marked as answer]
I don't know if this advice will help you with what you ultimately want to achieve, but here is my answer to your question.

I recommend that you check the following data.

https://github.com/OpenVSLAM-Community/openvslam/blob/6ce268c29539eba8ba624abc39fd5d178c85d3d0/src/openvslam/data/frame.h#L196-L204

https://github.com/OpenVSLAM-Community/openvslam/blob/6ce268c29539eba8ba624abc39fd5d178c85d3d0/src/openvslam/data/frame.h#L226-L227

https://github.com/OpenVSLAM-Community/openvslam/blob/6ce268c29539eba8ba624abc39fd5d178c85d3d0/src/openvslam/data/landmark.h#L126-L127

For example, referring to the same position in landmarks_ and keypts_ will yield a pair of a landmark and a keypoint. If there is no landmark corresponding to the keypoint, the landmark is nullptr.

**Reply by yyyutooo** [reply]
Thank you for your answer.
So, 'keypts_' represents the coordinates of the feature points in the image, and 'landmarks_' represents the location of the feature points on the map?

**Reply by ymd-stella** [reply]
Yes.

**Reply by yyyutooo** [reply]
I see.
How does 'referring to the same position in landmarks_ and keypts_' do it?
If I refer to each array with the same element number, will I get the corresponding data?

**Reply by ymd-stella** [reply]
Yes.

**Reply by yyyutooo** [reply]
Sorry.
I have tried to display the value of **'tracker->curr_frm_.landmarks_.at(i)'** every frame.
This resulted in the following values being displayed
```
0x7f07f0a837a0
0x7f07f2e380e0
0x7f07f29eef40
0
0
0
```
I understand that 0 means there is no value, but what does this hexadecimal value mean?
I am very sorry for this amateur question, but please help me.

**Reply by ymd-stella** [reply]
Pointer.

**Reply by yyyutooo** [reply]
I see that 'landmarks_' contains pointers, not coordinate data.
How can I get the coordinate data?
I'm sorry for repeating myself.

**Reply by ymd-stella** [reply]
`tracker->curr_frm_.landmarks_.at(i)->pos_w_`

**Reply by yyyutooo** [reply]
Thank you very much!
